### ISSUE TITLE
Universe unification drops constraints

### ISSUE THREAD

**Issue by coqbot** [coq/coq#4726]
Note: the issue was created automatically with bugzilla2github tool

Original bug ID: BZ#4726
From: @ppedrot
Reported version: 8.5
CC:   coq-bugs-redist@lists.gforge.inria.fr

**Comment by SkySkimmer** [comment]
@ppedrot can't reproduce, I get
```coq
Inj@{i j} : Type@{i} -> Type@{j} -> Type@{j}
(* i j |= i <= j
           *)
```
on 8.7.

**Comment by ppedrot** [comment]
Maybe it has been solved in the meantime. Is there a way to easily add a test?

**Comment by SkySkimmer** [comment]
You mean something like
```coq
Set Universe Polymorphism.

Record Inj@{i j} (A : Type@{i}) (B : Type@{j}) : Type@{j} := { inj : A }.

Fail Definition not_lost@{i j|j < i} := Inj@{i j}.
```
?

**Comment by ppedrot** [comment]
Yes, since we have a syntax for constraints it's way easier. Could you submit your test as a PR?

**Comment by SkySkimmer** [comment]
Actually just noticed we already have a test for this https://github.com/coq/coq/blob/master/test-suite/bugs/closed/4726.v.
git says fixed in c200c2b41e88dd7d4a1b9e90e0c35a7ed047309c.
